test/__mocks__/AudioNode.js

```diff
@@ -0,0 +1,36 @@
+class AudioNodeMock {
+    constructor (params) {
+        Object.assign(this, params);
+
+        this.connected = null;
+        this.connectedFrom = [];
+        this._testResult = [];
+    }
+
+    connect (node) {
+        this.connected = node;
+        node.connectedFrom.push(this);
+    }
+
+    disconnect () {
+        if (this.connected !== null) {
+            this.connectedFrom = this.connectedFrom.filter(connection => connection !== this);
+        }
+        this.connected = null;
+    }
+
+    _test (test, message, depth = 0) {
```

Comment:
Love the method body, but meh on the name, perhaps: `testConnectionPropagation` or oder or something?